### PR TITLE
Deprecate `service_url` methods in favour of `url`

### PR DESCRIPTION
**Pull request by peterzhu2118**
### Summary

#36729 added support for public services in Active Storage and deprecated `Blob#service_url` method for `Blob#url`. However, `Variant#service_url` and `Preview#service_url` were not updated to be consistent. This PR deprecates `Variant#service_url` and `Preview#service_url` and changes them to be `url` instead.

### CODE REVIEW

File: activestorage/app/models/active_storage/variant.rb [hunk 1]
```diff
@@ -73,18 +73,21 @@ def key
     "variants/#{blob.key}/#{Digest::SHA256.hexdigest(variation.key)}"
   end
 
-  # Returns the URL of the variant on the service. This URL is intended to be short-lived for security and not used directly
-  # with users. Instead, the +service_url+ should only be exposed as a redirect from a stable, possibly authenticated URL.
-  # Hiding the +service_url+ behind a redirect also gives you the power to change services without updating all URLs. And
-  # it allows permanent URLs that redirect to the +service_url+ to be cached in the view.
+  # Returns the URL of the blob on the service. This returns a permanent URL for public files, and returns a
```
Comment (gmcgibbon) [inline activestorage/app/models/active_storage/variant.rb]:
```suggestion
  # Returns the URL of the blob variant on the service. This returns a permanent URL for public files, and returns a
```

File: activestorage/app/models/active_storage/variant.rb [hunk 2]
```diff
@@ -73,18 +73,21 @@ def key
...
+  # short-lived URL for private files. Private files are for security and not used directly with users, instead,
```
Comment (gmcgibbon) [inline activestorage/app/models/active_storage/variant.rb]:
```suggestion
  # short-lived URL for private files. Private files are signed, and not for public use. Instead,
```

File: activestorage/app/models/active_storage/variant.rb [hunk 3]
```diff
@@ -73,18 +73,21 @@ def key
...
+  # the URL should only be exposed as a redirect from a stable, possibly authenticated URL. Hiding the
+  # URL behind a redirect also gives you the power to change services without updating all URLs.
```
Comment (gmcgibbon) [inline activestorage/app/models/active_storage/variant.rb]:
```suggestion
  # URL behind a redirect also allows you to change services without updating all URLs.
```

File: activestorage/app/models/active_storage/preview.rb [hunk 4]
```diff
@@ -56,15 +56,18 @@ def image
...
     if processed?
-      variant.service_url(**options)
+      variant.url(**options)
     else
       raise UnprocessedError
     end
   end
 
+  alias_method :service_url, :url
+  deprecate service_url: :url
```
Comment (bogdanvlviv) [inline activestorage/app/models/active_storage/preview.rb]: Was `service_url` added to Rails 6.0? If no, we can change this without deprecation.

File: activestorage/app/models/active_storage/variant.rb (hunk 1 above)
Comment (peterzhu2118) [reply activestorage/app/models/active_storage/variant.rb]: I updated `Variant` to reference the documentation for `Blob#url` to reduce duplication..

File: activestorage/app/models/active_storage/preview.rb (hunk 4 above)
Comment (peterzhu2118) [reply activestorage/app/models/active_storage/preview.rb]: I think `service_url` was available in Rails 5.2 as well: https://apidock.com/rails/v5.2.3/ActiveStorage/Variant/service_url

**Review by gmcgibbon — APPROVED** [review]

**Pull request merged**